File: Backend-C#/Infrastructure.MyShop/Interfaces/IRoleService.cs
using Infrastructure.MyShop.Models.DTO.RoleDTO;
using Infrastructure.MyShop.Services;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Infrastructure.MyShop.Interfaces
{
    public interface IRoleService
    {
        Task<ServiceResponse> GetAllRolesAsync();
        Task<ServiceResponse> GetRoleByIdAsync(long id);
        Task<ServiceResponse> CreateRoleAsync(RoleCreateDTO model);
        Task<ServiceResponse> EditRoleAsync(RoleEditDTO model);
        Task<ServiceResponse> DeleteRoleByIdAsync(long id);
    }
}

File: Backend-Java/src/main/java/org/example/Infrastructure/interfaces/IServices/IRoleService.java
package org.example.Infrastructure.interfaces.IServices;

import org.example.Infrastructure.dto.roleDTO.*;

import java.util.List;

public interface IRoleService {
    RoleItemDTO create(RoleCreateDTO model);
    List<RoleItemDTO> getAll();
    RoleItemDTO edit(RoleEditDTO model);
    void deleteById(int id);
    RoleItemDTO getById(int id);
}
